autode/plotting.py

```diff
@@ -376,3 +390,76 @@ def error_on_stationary_points(x, energies):
     energy_difference = energies - np.array(energies_at_stationary_points)
 
     return np.sum(np.square(energy_difference))
+
+
+def plot_optimiser_profile(
+    history: "_OptimiserHistory",
+    plot_energy: bool,
+    plot_rms_grad: bool,
+    filename: str,
+):
+    """
+    Plot the energy and RMS gradient profile from an optimiser history.
+    Skips plotting of points where energy/grad is not available
+
+    -------------------------------------------------------------------------
+    Args:
+        history (_OptimiserHistory): History (list) of coordinate objects
+        plot_energy (bool): Whether to plot energy or not
+        plot_rms_grad (bool): Whether to plot rms grad or not
+        filename (str): Name of plotted file
+    """
+    if not (plot_energy or plot_rms_grad):
+        logger.error(
+            "Must plot either energies or RMS gradients for an"
+            "optimiser profile"
+        )
+        return None
+
+    import matplotlib.pyplot as plt
+    from matplotlib.ticker import MaxNLocator
+
+    x_axis = [i + 1 for i in range(len(history))]  # starts at 0
+    energies = []
+    rms_grads = []
+    for coord in history:
+        if coord.e is not None:
+            energies.append(coord.e.to("Ha"))
+        else:
+            energies.append(np.nan)
+
+        if coord.g is not None:
+            rms = np.sqrt(np.average(np.square(coord.to("cart").g)))
+            rms_grads.append(rms)
+        else:
+            rms_grads.append(np.nan)
+
+    fig, ax = plt.subplots()
+
+    if plot_energy:
+        ax.plot(
+            x_axis, energies, "o-", color="C0", label="Electronic energy"
+        )  # blue
+        ax.set_xlabel("Optimiser step")
+        ax.set_ylabel("Electronic energy / Ha")
+
+    ax.set_xlim(left=0.5)
+    ax.xaxis.set_major_locator(
+        MaxNLocator(nbins="auto", steps=[1, 2, 2.5, 5, 10], integer=True)
+    )
+
+    if plot_rms_grad:
+        if not plot_energy:
+            ax2 = ax
+        else:
+            ax2 = ax.twinx()  # plot on a different axis in same fig
```

Comment:
clearer to me as a ternary
```suggestion
        ax2 = ax.twinx() if plot_energy else ax
```